changelog.lua

```diff
@@ -0,0 +1,374 @@
+}
+local SCOPE_PRIORITY = { -- smallest on top
+  Performance = 10,
+  Configuration = 20,
+  Core = 30,
+  PDK = 40,
+  Plugin = 50,
+  ["Admin API"] = 60,
+  Clustering = 70,
+  Default = 100, -- default priority
+}
+
+setmetatable(SCOPE_PRIORITY, {
+  __index = function()
+    return rawget(SCOPE_PRIORITY, "Default") - 1
+  end
+})
+
+local function table_keys(t)
+  if type(t) ~= "table" then
+    return t
+  end
+  local keys = {}
+  for k, _ in pairs(t) do
+    table.insert(keys, k)
+  end
+  return keys
+end
+
+local function parse_github_ref(system, reference_type, references)
+  if references == nil or references == lyaml.null then
+    return nil
+  end
+  local parsed_references = {}
+  for i, ref in ipairs(references or {}) do
+    local repo = REPOS[system]
+    local ref_no = tonumber(ref) -- treat ref as number string first
+    local name = "#" .. ref
+    if not ref_no then -- ref is not a number string
+      local parts = pl_stringx.split(ref, ":")
+      repo = parts[1]
+      ref_no = parts[2]
+      name = pl_stringx.replace(tostring(ref), ":", " #")
+    end
+    parsed_references[i] = {
+      id = ref_no,
+      name = name,
+      link = string.format(GITHUB_REFERENCE[reference_type], repo, ref_no),
+    }
+  end
+  return parsed_references
+end
+
+
+local function parse_jiras(jiras)
+  local jira_items = {}
+  for i, jira in ipairs(jiras or {}) do
+    jira_items[i] = {
+      id = jira,
+      link = JIRA_BASE_URL .. jira
+    }
+  end
+  return jira_items
+end
+
+
+local function is_yaml(filename)
+  return pl_stringx.endswith(filename, ".yaml") or
+    pl_stringx.endswith(filename, ".yml")
+end
+
+local function is_empty_table(t)
+  return t == nil or next(t) == nil
+end
+
+local function compile_template(data, template)
+  local compile_env = {
+    _escape = ">",
+    _brackets = "{}",
+    _debug = true,
+    pairs = pairs,
+    ipairs = ipairs,
+    tostring = tostring,
+    is_empty_table = is_empty_table,
+  }
+
+  compile_env = pl_tablex.merge(compile_env, data, true) -- union
+  local content, err = pl_template.substitute(template, compile_env)
+  if not content then
+    return nil, "failed to compile template: " .. err
+  end
+
+  return content
+end
+
+local function absolute_path(...)
+  local path = CHANGELOG_PATH
+  for _, p in ipairs({...}) do
+    path = pl_path.join(path, p)
+  end
+  return path
+end
+
+local function collect_files(folder)
+  local files
+  if pl_path.exists(folder) then
+    files = assert(pl_dir.getfiles(folder))
+    if files then
+      table.sort(files)
+    end
+  end
+  local sorted_files = {}
+  for _, filename in ipairs(files or {}) do
+    if is_yaml(filename) then
+      table.insert(sorted_files, filename)
+    end
+  end
+
+  return sorted_files
+end
+
+
+local function fetch_commit_context(filename)
+  local ctx = {}
+  filename = string.sub(filename, 6)
+  --print(filename)
+  local commit_url = "https://api.github.com/repos/kong/kong/commits?path=%s"
```

Comment:
Should the `kong` repo name be configurable? We have other repos we might use as well.